java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1157,62 +1157,61 @@ private void handleImageBuildData(ServerAction serverAction, JsonElement jsonRes
+                        action.getName() + " in handleImageBuildData");
+                return;
+            }
+            Optional<ImageProfile> profileOpt =
+                    ImageProfileFactory.lookupById(imageProfileId.get());
+            profileOpt.ifPresent(p -> p.asKiwiProfile().ifPresent(kiwiProfile -> {
+                serverAction.getServer().asMinionServer().ifPresent(minionServer -> {
+                    // Download the built Kiwi image to SUSE Manager server
+                    OSImageInspectSlsResult.Bundle bundleInfo =
+                            Json.GSON.fromJson(jsonResult, OSImageBuildSlsResult.class)
+                                    .getKiwiBuildInfo().getChanges().getRet().getBundle();
+                    infoOpt.ifPresent(info -> info.setChecksum(
+                            ImageInfoFactory.convertChecksum(bundleInfo.getChecksum())));
+                    MgrUtilRunner.ExecResult collectResult = systemQuery
+                            .collectKiwiImage(minionServer, bundleInfo.getFilepath(),
+                                    OSImageStoreUtils.getOsImageStorePath() + kiwiProfile.getTargetStore().getUri())
+                            .orElseThrow(() -> new RuntimeException("Failed to download image."));
+                    if (collectResult.getReturnCode() != 0) {
+                        serverAction.setStatus(ActionFactory.STATUS_FAILED);
+                        serverAction.setResultMsg(StringUtils
+                                .left(printStdMessages(collectResult.getStderr(), collectResult.getStdout()),
+                                        1024));
+                    }
                 });
-            }, () -> {
-                LOG.error("Details not found in ImageBuildAction");
-                LOG.error("Name is: " + action.getName());
-                LOG.error("Action ID is: " + action.getId());
-                LOG.error("Earliest action was: " + action.getEarliestAction());
-                LOG.error("Scheduler User is: " + action.getSchedulerUser());
+            }));
+            ImageInspectAction iAction = ActionManager.scheduleImageInspect(
+                    action.getSchedulerUser(),
+                    action.getServerActions()
+                            .stream()
+                            .map(ServerAction::getServerId)
+                            .collect(Collectors.toList()),
+                    Optional.of(action.getId()),
+                    details.get().getVersion(),
+                    profileOpt.map(ImageProfile::getLabel).orElse(null),
+                    profileOpt.map(ImageProfile::getTargetStore).orElse(null),
+                    Date.from(Instant.now())
+            );
+            try {
+                TASKOMATIC_API.scheduleActionExecution(iAction);
+            }
+            catch (TaskomaticApiException e) {
+                LOG.error("Could not schedule image inspection");
```

Comment:
Why not simply `LOG.error("Could not schedule image inspection", e);`

Reply:
agree